### PR TITLE
prowgen: stop tolerating changes to `optional` fields

### PR DESCRIPTION
**Pull request by petr-muller**
...except for images jobs for which do not have syntax in ci-operator config

Once we pull all manual changes to ci-operator configs, we can stop tolerating these changes. One step closer to fully generated Prowjob configuration.

Needs https://github.com/openshift/release/pull/24629 for breaking changes to pass (we only want to merge this once breaking changes passes)

/hold
/cc @droslean @openshift/test-platform

### CODE REVIEW

**Comment by petr-muller** [comment]
/test e2e

It's all failures to reach release-controllers, not our code :(

**Review by hongkailiu — APPROVED** [review]

**Comment by petr-muller** [comment]
note to myself: I want to merge this only after it passes `breaking-changes`

**Comment by petr-muller** [comment]
:heart: perfect!

/hold cancel

**Pull request merged**
